roles/openshift_hosted/tasks/router/router.yml

```diff
@@ -10,7 +10,7 @@
 
 - name: set_fact replicas
   set_fact:
-    replicas: "{{ openshift.hosted.router.replicas | default(router_nodes.results.results[0]['items'] | length) }}"
+    replicas: "{{ openshift.hosted.router.replicas|default(None) | get_router_replicas(router_nodes) }}"
```

Comment:
@detiber, I think that since the first parameter takes precedence I'd prefer it as the one being piped.  The integer you refer to isn't always set.